### DISCUSSION TITLE
Issues with mesh generation in mri2mesh pipeline

### DISCUSSION THREAD

**Discussion by SeSodesa** [sampsapursiainen/zeffiro_interface#177] [Q&A]
I've been reworking the `mri2mesh` pipeline to make it work with the changes made over the past few months. After remembering that certain import files need to be in their relevant folders, everything is almost as it should be, but I am currently getting the following error message when trying to create a head model from MRI data `recon-all`ed with FreeSurfer:

```matlab
Arrays have incompatible sizes for this operation.

Error in zef_inflate_surfaces (line 96)
d_vec = p(ones_vec_nearest,:)  -
nodes_tri_ref(tri_ref(nearest_neighbor_ind(block_ind(k),:),1),:);

Error in zef_smoothing_step (line 272)
nodes = zef_inflate_surfaces(zef, nodes,tetra,submesh_ind_1(domain_labels));

Error in zef_postprocess_fem_mesh (line 113)
zef_smoothing_step;

Error in zef_create_finite_element_mesh (line 12)
zef = zef_postprocess_fem_mesh(zef);

Error in zeffiro_interface (line 517)
        eval(args.run_script);
```

The script given to Zeffiro can be seen here: <https://github.com/sampsapursiainen/zeffiro_interface/blob/update-mri2mesh/scripts/studies/mri2mesh/mri2mesh.sh#L223>. Does something in it need to change to reflect the new state of things, or is there some unaccounted-for edge case in the mesh generation routine?

**Comment by mshavliuk** [comment]
@SeSodesa is it still a relevant discussion?

**Reply by SeSodesa** [reply]
Yes and no. The recent merge of a related pull request #193 made the `fs2zef` step slightly more effortless (I hope), but this refers to the mesh construction step, which was not (yet) added to the new `mri2mesh` package. The `mri2mesh.fs2zef` function might be extended, such that if `meshgen_output_path` and other meshing-related options are provided via optional (keyword) arguments, `zeffiro_interface` is invoked and the resulting project file is saved to the target location.

Now, somebody would just have to document the mesh generation -related `zef` fields, so that they could be set programmatically to sensible values. I certainly don't know all of them.

**Reply by SeSodesa** [reply]
For example, what do the numbers in the array

    project_struct.refinement_surface_compartments = [10 -1 1 18 17];

we discussed earlier refer to? Compartment labels? But if so, what is that `-1` doing there? Does the order matter? What are the possible values for them. I guess the source code of **Mesh tool** and **Menu tool > Forward and inverse processing options** windows could be looked at, and at least some of the information, such as default values might be extracted from there.

**Reply by SeSodesa** [reply]
The source file which defines the default values of the fields of forward and inverse options seems to be in `m/zef_init_forward_and_inverse_options`.

**Reply by SeSodesa** [reply]
Actually, `m/zef_update_forward_and_inverse_options.m` might be even more informative, as it sets many of the project struct values when text fields and such in the GUI are updated.
